src/Surfnet/Stepup/Identity/Entity/VerifiedSecondFactor.php

```diff
@@ -89,10 +91,10 @@ public static function create(
         SecondFactorType $type,
         SecondFactorIdentifier $secondFactorIdentifier,
         DateTime $registrationRequestedAt,
-        $registrationCode
+        string $registrationCode
     ) {
-        if (!is_string($registrationCode)) {
-            throw InvalidArgumentException::invalidType('string', 'registrationCode', $registrationCode);
+        if (!preg_match('/^[A-Z0-9]{8}$/i', $registrationCode)) {
```

Comment:
I love to have a more specific input validation here. It might be a good idea to introduce a registration code value object that is tasked with validating the code? That's more in line with the way the domain is set up. For now we kept it primitive because we did not realy bother with elaborate validation.

Reply:
If you want to go the extra mile, then please be my guest. I'm not going to solve the world's problems here ;)

Reply:
In this case I think it's just a first guard on "sane" input as a defense-in-depth: does this even look like a registration code? The actual validity of the given code will be verified later obviously. So I can imagine that it's just fine to do the basic check that Tim implemented.